core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go

```diff
@@ -162,10 +163,11 @@ func (r *EvmRegistry) streamsLookup(ctx context.Context, checkResults []ocr2keep
 func (r *EvmRegistry) doLookup(ctx context.Context, wg *sync.WaitGroup, lookup *StreamsLookup, i int, checkResults []ocr2keepers.CheckResult, lggr logger.Logger) {
 	defer wg.Done()
 
-	state, reason, values, retryable, err := r.doMercuryRequest(ctx, lookup, lggr)
+	state, reason, values, retryable, ri, err := r.doMercuryRequest(ctx, lookup, checkResults[i].WorkID+"|"+fmt.Sprintf("%d", lookup.block), lggr)
```

Comment:
nit. having key generation in a separate function 'pluginRetryKey' will make it more explicit rather than being hidden inside a function call

Reply:
done